object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java

```diff
@@ -49,6 +63,23 @@ public Void visitAnnotation(final AnnotationTree node, final Void p) {
   @Override
   public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
 
+    if (!isAssignedToLocal(this.getCurrentPath()) && !atypeFactory.returnsThis(node)) {
+      TypeMirror returnType = TreeUtils.typeOf(node);
+
+      if (hasAlwaysCall(returnType)) {
+        TypeElement eType = TypesUtils.getTypeElement(returnType);
+        AnnotationMirror alwaysCallAnno = atypeFactory.getDeclAnnotation(eType, AlwaysCall.class);
+
+        String alwaysCallAnnoVal =
+            AnnotationUtils.getElementValue(alwaysCallAnno, "value", String.class, false);
+        List<String> currentCalledMethods = getCalledMethodAnnotation(node);
+
+        if (!currentCalledMethods.contains(alwaysCallAnnoVal)) {
```

Comment:
This should also probably use `isSubtype` from the type factory rather than implement the check directly.